cmd/iam-object-store.go

```diff
@@ -433,6 +433,13 @@ var (
 func (iamOS *IAMObjectStore) listAllIAMConfigItems(ctx context.Context) (map[string][]string, error) {
 	res := make(map[string][]string)
 
+	ignoredItems := map[string]struct{}{
+		minioConfigPrefix + SlashSeparator + minioConfigFile: struct{}{},
+		tierConfigPath:                 struct{}{},
+		SlashSeparator + iamFormatFile: struct{}{},
+		iamFormatFile:                  struct{}{},
+	}
+
 	for item := range listIAMConfigItems(ctx, iamOS.objAPI, iamConfigPrefix) {
```

Comment:
When we are listing only `iamConfigPrefix` (= `minioConfigPrefix + "/iam"`), it shouldn't list `tierConfigPath`:

```
var tierConfigPath = path.Join(minioConfigPrefix, tierConfigFile)
```

right?

Even the history files, why are they getting listed?

Reply:
Because we are listing config/

Reply:
It's the same reason config.json gets listed @donatello